command/pr_create.go

```diff
@@ -15,6 +15,39 @@ import (
 	"github.com/spf13/cobra"
 )
 
+type defaults struct {
+	Title string
+	Body  string
+}
+
+func computeDefaults(baseRef, headRef string) (defaults, error) {
+	commits, err := git.Commits(baseRef, headRef)
+	if err != nil {
+		return defaults{}, err
+	}
+
+	out := defaults{}
+
+	if len(commits) == 1 {
+		out.Title = commits[0].Title
+		body, err := git.CommitBody(commits[0].Sha)
+		if err != nil {
+			return defaults{}, err
+		}
+		out.Body = body
+	} else {
+		out.Title = headRef // TODO format or something?
+
+		body := fmt.Sprintf("---\n%d commits:\n\n", len(commits))
```

Comment:
What if instead of an `<hr>` separator and the `{N} commits:` opening, we had none of that; i.e. having a list of commits simply appear as a Markdown list? That way the pre-rendered pull request body could immediately look good for submitting, or maybe the person will choose to take some lines out if they are not really relevant for PR description.

I'm thinking that because, if someone submits the `---\n%d commits:\n` text as-generated without editing, what results is a dry, mechanical listing of git commits embedded in the pull request body, whereas every PR already comes with a list of git commits attached with it: the "Commits" tab.

If we make the default text more human-like, I think it might result in pull request bodies that are nicer to read.

Reply:
i like that idea! i was hoping to get @ampinsk 's input, too.